### DISCUSSION TITLE
Icons of `Checkbox` are not loaded properly

### DISCUSSION THREAD

**Discussion by daya0576** [zauberzeug/nicegui#2834] [Q&A]
### Question

# Problem
When the page loaded the first time or without cache, is it possible to prevent unexpected text before the icon is unable to display?

<img width="300" alt="image" src="https://github.com/zauberzeug/nicegui/assets/6239652/869d5dd3-46bc-4a79-93ef-cd0097351bb8">

# Code example
https://github.com/daya0576/beaverhabits/blob/master/beaverhabits/frontend/components.py#L34
```python
c = CheckBox()
c.props('checked-icon="sym_r_done" unchecked-icon="sym_r_close" keep-color')
```

Thanks in advance for the awesome framework!

**Comment by falkoschindler** [comment]
Thanks for bringing this up, @daya0576!

I can reproduce the problem running the following code with a throttled network and disabled browser cache:
```py
ui.checkbox().props('checked-icon=sym_r_done unchecked-icon=sym_r_close')
```

I'm not sure if there's a way to fix it. It also happens with a pure Quasar app:

```html
<html>
  <head>
    <link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700,900|Material+Icons" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/npm/quasar@2.15.2/dist/quasar.prod.css" rel="stylesheet" type="text/css" />
  </head>
  <body>
    <div id="q-app">
      <q-checkbox v-model="checked" checked-icon="check" unchecked-icon="close" />
    </div>
    <script src="https://cdn.jsdelivr.net/npm/vue@3/dist/vue.global.prod.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/quasar@2.15.2/dist/quasar.umd.prod.js"></script>
    <script>
      const app = Vue.createApp({ setup: () => ({ checked: Vue.ref(false) }) });
      app.use(Quasar);
      app.mount("#q-app");
    </script>
  </body>
</html>
```

**Comment by daya0576** [comment]
@falkoschindler Thanks for your quick answer. 

The root cause of this issue is likely that Google font files are loaded after the main page has been generated, the font file is more than 200KB!

So my current workaround is to put raw SVG data to runtime to fix this issue.

For example:
```python
# icons.py
SVG_PREFIX = "img:data:image/svg+xml;charset=utf8"
# pay attention to using single quotes
DONE = f"{SVG_PREFIX},<svg xmlns='http://www.w3.org/2000/svg' height='24' viewBox='0 -960 960 960' width='24' fill='rgb(103,150,207	)'><path d='m382-354 339-339q12-12 28.5-12t28.5 12q12 12 12 28.5T778-636L410-268q-12 12-28 12t-28-12L182-440q-12-12-11.5-28.5T183-497q12-12 28.5-12t28.5 12l142 143Z'/></svg>"

# components.py
ui.checkbox().props(f'checked-icon="{icons.DONE}" unchecked-icon="sym_r_close" keep-color')
```
